flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/Trigger.java

```diff
@@ -18,28 +18,38 @@
  */
 package org.apache.iceberg.flink.maintenance.api;
 
+import javax.annotation.Nullable;
 import org.apache.flink.annotation.Internal;
+import org.apache.iceberg.actions.ActionResult;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 
 public class Trigger {
   private final long timestamp;
   private final Integer taskId;
   private final boolean isRecovery;
+  private final @Nullable ActionResult actionResult;
 
-  private Trigger(long timestamp, Integer taskId, boolean isRecovery) {
+  private Trigger(
+      long timestamp, Integer taskId, boolean isRecovery, @Nullable ActionResult actionResult) {
     this.timestamp = timestamp;
     this.taskId = taskId;
     this.isRecovery = isRecovery;
+    this.actionResult = actionResult;
```

Comment:
`Trigger` loses some of its purity if we use it for something else (storing the result of a trigger). Could we add a new type `TriggerResult`?